src/main/java/org/opensearch/ad/rest/RestAnomalyDetectorJobAction.java

```diff
@@ -16,60 +16,138 @@
+        this.extensionsRunner = extensionsRunner;
+        this.client = client;
+        this.clusterService = extensionsRunner.getSdkClusterService();
+        this.settings = extensionsRunner.getEnvironmentSettings();
         this.requestTimeout = REQUEST_TIMEOUT.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(REQUEST_TIMEOUT, it -> requestTimeout = it);
+        this.jobDetailsAreRegistered = false;
     }
 
-    @Override
     public String getName() {
         return AD_JOB_ACTION;
     }
 
     @Override
-    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+    public List<RouteHandler> routeHandlers() {
+        return ImmutableList
+            .of(
+                // start AD Job
+                new RouteHandler(
+                    RestRequest.Method.POST,
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, DETECTOR_ID, START_JOB),
+                    handleRequest
+                )
+            );
+    }
+
+    private Function<RestRequest, ExtensionRestResponse> handleRequest = (request) -> {
+        try {
+            return prepareRequest(request);
+        } catch (Exception e) {
+            return exceptionalRequest(request, e);
+        }
+    };
+
+    private void registerJobDetails() throws IOException {
+
+        XContentBuilder requestBody = JsonXContent.contentBuilder();
+        requestBody.startObject();
+        requestBody.field(GetJobDetailsRequest.JOB_INDEX, AnomalyDetectorJob.ANOMALY_DETECTOR_JOB_INDEX);
+        requestBody.field(GetJobDetailsRequest.JOB_TYPE, AnomalyDetectorExtension.AD_JOB_TYPE);
+        requestBody.field(GetJobDetailsRequest.JOB_PARAMETER_ACTION, ADJobParameterAction.class.getName());
+        requestBody.field(GetJobDetailsRequest.JOB_RUNNER_ACTION, ADJobRunnerAction.class.getName());
+        requestBody.field(GetJobDetailsRequest.EXTENSION_UNIQUE_ID, extensionsRunner.getUniqueId());
+        requestBody.endObject();
+
+        Request request = new Request("PUT", String.format(Locale.ROOT, "%s/%s", JobSchedulerPlugin.JS_BASE_URI, "_job_details"));
+        request.setJsonEntity(Strings.toString(requestBody));
+
+        Response response = client.performRequest(request);
+        this.jobDetailsAreRegistered = RestStatus.fromCode(response.getStatusLine().getStatusCode()) == RestStatus.OK ? true : false;
+        LOG.info("Job Details Registered : " + jobDetailsAreRegistered);
+    }
+
+    protected ExtensionRestResponse prepareRequest(RestRequest request) throws IOException {
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
 
+        // Ensure job details are registered with Job Scheduler prior to creating a job
+        if (!jobDetailsAreRegistered) {
```

Comment:
For my understanding. How did we ensure that job details are registered before in AD plugin?

Reply:
For plugins, job details are registered during the bootstrap of opensearch through the `JobSchedulerExtension` extension points [here](https://github.com/opensearch-project/anomaly-detection/blob/fd2fd5d5f63c9004b9f265e3c1e4475282447da8/src/main/java/org/opensearch/ad/AnomalyDetectorPlugin.java#L1012). For extensions, this wont work, since extensions are initialized after the bootstrap of opensearch and can come up and down at anytime (hotswap). Additionally, an extension may have the need to register more than one type of job. 

To support this, the job scheduler communication mechanism has introduced a REST API that an extension may call at any time. The only caveat is that these details must be sent to Job Scheduler prior to the initialization of the corresponding job index, which is why we have sent this request prior to the processing of the start detector request.

Reply:
Got it. Is this the only place we should check whether the job details are registered?

Reply:
Yes, since Job Scheduler is used only for real time analysis, which is triggered by `startdetector`.